File: fluterfront/lib/models/producto.dart
class Producto {
  final int id;
  final String nombre;
  final String costo;
  final int categoriaId;
  final String? descripcion;

  Producto({
    required this.id,
    required this.nombre,
    required this.costo,
    required this.categoriaId,
    this.descripcion,
  });

  factory Producto.fromJson(Map<String, dynamic> json) {
    return Producto(
      id: json['id'],
      nombre: json['nombre'],
      costo: json['costo'],
      categoriaId: json['categoriaId'],
      descripcion: json['descripcion'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'costo': costo,
      'categoriaId': categoriaId,
      'descripcion': descripcion,
    };
  }
}

File: ProyectoIntegradorAndroid/app/src/main/java/com/example/sistemaventas/Producto.kt
data class Producto(
    val id: Int,
    val nombre: String,
    val costo: String,
    val categoriaId: Int,
    val tasaIGV: String?,
    val descripcion: String,
    val codigoBarras: String?,
    val cantidadStock: String?,
    val disponible: String?,
    val proveedor: String?,
    val categoria: Categoria?  // Incluye la clase Categoria
)

data class Categoria(
    val id: Int,
    val titulo: String,
    val descripccion: String
)
